core/server/api/posts.js

```diff
@@ -55,7 +55,27 @@ posts = {
          * @returns {Object} options
          */
         function modelQuery(options) {
-            return dataProvider.Post.findPage(options);
+            var requestedColumns = [];
+
+            // 'url' and 'author' depend on other columns. Make sure to load them.
+            if (_.isArray(options.columns)) {
+                requestedColumns = _.clone(options.columns);
+                if (_.includes(options.columns, 'url') || _.includes(options.columns, 'author')) {
+                    options.columns.push('id', 'published_at', 'slug', 'author_id');
```

Comment:
there are two ways: either we fetch by default almost everything and pick what the user has requested for (maybe a performance issue) or we always fetch a minimum set of attributes per model (depends what each model needs minimum to fulfil its needs).

in this case: there is a somewhere a piece of logic in the `toJSON` fn to generate a dynamic URL, which depends on a set of attributes. in my opinion we shouldn't add a another piece of logic somewhere in a controller, so that the `toJSON` will not return undefined.